docs/snapshot-appr-registry.md

```diff
@@ -0,0 +1,152 @@
+
+## Prerequisites
+
+- Linux
+- A version `oc` that has the `oc catalog build`, such as [this one for OCP `4.3`](https://openshift-release-artifacts.svc.ci.openshift.org/4.3.0-0.nightly-2019-11-13-233341/openshift-client-linux-4.3.0-0.nightly-2019-11-13-233341.tar.gz)
+- A container image registry that supports [Docker v2-2](https://docs.docker.com/registry/spec/manifest-v2-2/)
+- [grpcurl](https://github.com/fullstorydev/grpcurl) (optional, for testing)
+
+## Setup
+
+Users should authenticate with the target image registry. By default, `oc adm catalog build` uses `~/.docker/config.json` to determine credentials. 
+
+## Taking a snapshot
+
+Snapshot `redhat-operators` as `quay.io/my/redhat-operators:snapshot-0`.
+
+```sh
+$ oc adm catalog build --appregistry-endpoint https://quay.io/cnr --appregistry-org redhat-operators --to=quay.io/my/redhat-operators:snapshot-0
+INFO[0013] loading Bundles                               dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605
+INFO[0013] directory                                     dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605 file=manifests-829192605 load=bundles
+INFO[0013] directory                                     dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605 file=3scale-operator load=bundles
+INFO[0013] found csv, loading bundle                     dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605 file=3scale-operator.v0.3.0.clusterserviceversion.yaml load=bundles
+INFO[0013] loading bundle file                           dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605/3scale-operator file=3scale-operator.package.yaml load=bundle
+INFO[0013] loading bundle file                           dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605/3scale-operator file=3scale-operator.v0.3.0.clusterserviceversion.yaml load=bundle
+...
+Uploading ... 244.9kB/s
+Pushed sha256:f73d42950021f9240389f99ddc5b0c7f1b533c054ba344654ff1edaf6bf827e3 to quay.io/my/redhat-operators:snapshot-0
+```
+
+Sometimes bad manifests are accidentally introduced into Red Hat's catalogs, when this happens you may see some errors.
+
+```sh
+...
+INFO[0014] directory                                     dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605 file=4.2 load=package
+W1114 19:42:37.876180   34665 builder.go:141] error building database: error loading package into db: fuse-camel-k-operator.v7.5.0 specifies replacement that couldn't be found
+Uploading ... 244.9kB/s
+...
+```
+
+These errors are usually non-fatal, and if the operator package mentioned doesn't contain an operator you plan to install or a dependency of one, then they can be ignored.
+
+## Testing a snapshot
+
+You can validate snapshot content by running it as a container and querying its gRPC API:
```

Comment:
This section seems like it could just be generically referenced in the `validation` doc?

Also, do we want to try to promote telling people to test their catalogs by starting them locally and querying with `grpcurl`? Since that is indirected via the packageserver anyway, it seems like an implementation detail. My feeling is that the more obvious way for users to grok this is: just create a catalogsource with the image and see if you can install the operators you care about.

Reply:
> This section seems like it could just be generically referenced in the validation doc?

Yeah. Do you think we should make the validation doc w/ just this section in this PR, or merge it like this and denormalize when the larger validation doc is added?

> Also, do we want to try to promote telling people to test their catalogs by starting them locally and querying with grpcurl? Since that is indirected via the packageserver anyway, it seems like an implementation detail. My feeling is that the more obvious way for users to grok this is: just create a catalogsource with the image and see if you can install the operators you care about.

Yes, I think this is a useful way to see if a catalog is working before attempting to run it on a cluster. Additionally, packageserver only tests a subset of a catalog's APIs; for example, you can't tell if the update graph for a package is what you expect. I think this can be a huge time-saver. 

Since we decided to standardize on SQLite, we can also have some docs that use DBA tooling to inspect catalogs.